dev/TabView/TabView.cpp

```diff
@@ -227,6 +237,20 @@ void TabView::OnSelectedItemPropertyChanged(const winrt::DependencyPropertyChang
     UpdateSelectedItem();
 }
 
+void TabView::OnTabStripFooterPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    if (const auto oldFooter = args.OldValue().try_as<winrt::FrameworkElement>())
+    {
+        oldFooter.UnregisterPropertyChangedCallback(winrt::FrameworkElement::MinWidthProperty(), m_footerMinWidthProperyChangedToken);
+        oldFooter.UnregisterPropertyChangedCallback(winrt::FrameworkElement::WidthProperty(), m_footerWidthProperyChangedToken);
+    }
+    if (const auto newFooter = args.NewValue().try_as<winrt::FrameworkElement>())
+    {
+        m_footerMinWidthProperyChangedToken = newFooter.RegisterPropertyChangedCallback(winrt::FrameworkElement::MinWidthProperty(), { this,&TabView::OnComponentSizeChanged });
```

Comment:
can these be done using a revoker ?

Reply:
I don't think that's possible. We need to listen to more or less generic dependency property changes, there is no dedicated (Min)WidthPropertyChanged event, so we need to use that route. Unfortunately, RegisterPropertyChangedCallback only works through tokens.

Reply:
@chingucoding please use the cpp winrt helper RegisterPropertyChanged [https://github.com/microsoft/microsoft-ui-xaml/blob/cc0d193295964fc0a94d78e39acacf4cb1f0e033/dev/inc/CppWinRTHelpers.h#L280](https://github.com/microsoft/microsoft-ui-xaml/blob/cc0d193295964fc0a94d78e39acacf4cb1f0e033/dev/inc/CppWinRTHelpers.h#L280)

---
In reply to: [528202327](https://github.com/microsoft/microsoft-ui-xaml/pull/3531#discussion_r528202327) [](ancestors = 528202327)

Reply:
Sure, wish I knew that existed earlier. Is fixed now.

I also noticed that a lot of the code uses the token pattern to do this, should we updated all occurrences to the helper instead of a token?